tests/TestPasskeys.cpp

```diff
@@ -0,0 +1,474 @@
+/*
+ *  Copyright (C) 2023 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "TestPasskeys.h"
+#include "browser/BrowserCbor.h"
+#include "browser/BrowserMessageBuilder.h"
+#include "browser/BrowserService.h"
+#include "crypto/Crypto.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QTest>
+#include <botan/sodium.h>
+
+using namespace Botan::Sodium;
+
+QTEST_GUILESS_MAIN(TestPasskeys)
+
+// Register request
+// clang-format off
+const QString PublicKeyCredentialOptions =
+    QString("{"
+                "\"attestation\": \"none\","
+                "\"authenticatorSelection\": {"
+                    "\"residentKey\": \"preferred\","
+                    "\"requireResidentKey\": false,"
+                    "\"userVerification\": \"required\""
+                "},"
```

Comment:
You could use a raw string literal for these. Then you don't have to escape all the quotes.